services/merak-agent/agent.go

```diff
@@ -87,38 +102,38 @@ func main() {
 	}()
 
 	pb.RegisterMerakAgentServiceServer(gRPCServer, &handler.Server{})
-	log.Printf("Starting gRPC server. Listening at %v", lis.Addr())
+	handler.MerakLogger.Info("Starting gRPC server. Listening at ", "addr", lis.Addr().String())
 	if err := gRPCServer.Serve(lis); err != nil {
-		log.Fatalf("failed to serve: %v\n", err)
+		handler.MerakLogger.Fatal("failed to serve \n", "err", err)
 	}
 }
 
 func startPlugin() {
 	if len(os.Args) < 3 {
-		log.Fatal("Not enough arguments")
+		handler.MerakLogger.Fatal("Not enough arguments")
 	}
 	remote_server := os.Args[1]
 	if net.ParseIP(remote_server) == nil {
-		log.Fatalf("Invalid IP address %s\n", remote_server)
+		handler.MerakLogger.Fatal("Invalid IP address " + remote_server)
 	}
 	handler.RemoteServer = remote_server
 	remote_port := os.Args[2]
 	remote_port_int, err := strconv.Atoi(os.Args[2])
 
 	if err != nil {
-		log.Fatalf("Port: %d is not a valid number!\n", remote_port_int)
+		handler.MerakLogger.Fatal("Port: is not a valid number!\n", remote_port_int)
 	}
 	if remote_port_int > constants.MAX_PORT || remote_port_int < constants.MIN_PORT {
-		log.Fatalf("Port: %d is not within a valid range!\n", remote_port_int)
+		handler.MerakLogger.Fatal("Port: is not within a valid range!\n", remote_port_int)
 	}
 	cmdString := "service rsyslog restart && /etc/init.d/openvswitch-switch restart && /merak-bin/AlcorControlAgent -d -a " + remote_server + " -p " + remote_port
-	log.Println("Executing command " + cmdString)
```

Comment:
@phudtran this `cmdString` should get from `Service Config` from scenario manager:
```
{
            "cmd": "/root/alcor-control-agent/build/bin/AlcorControlAgent",
            "name": "aca-cmd",
            "parameters": [
                "-d",
                "-a 10.213.43.111",
                "-p 30014"
            ],
            "return_code": [
                0
            ],
            "return_string": [
                ""
            ],
            "url": "",
            "when_to_run": "INIT",
            "where_to_run": "AGENT"
        }
```

Reply:
I'm currently parsing the remote_server and remote_port that Topo passes in as cmdline args from scenario manager's service config. We can change this to the entire command, but will also need a change from merak-topo side.